File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/presentation/common/ConnectionUtils.java
package com.crpr.androidcinema.presentation.common;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

/**
 * Created by DEATH_STAR on 21/01/2017.
 */

public class ConnectionUtils {

    public static boolean isConnectionAvailable(Context context) {
        int[] networkTypes = {ConnectivityManager.TYPE_MOBILE,
                ConnectivityManager.TYPE_WIFI};
        try {
            ConnectivityManager connectivityManager =
                    (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
            for (int networkType : networkTypes) {
                NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
                if (activeNetworkInfo != null &&
                        activeNetworkInfo.getType() == networkType)
                    return true;
            }
        } catch (Exception e) {
            return false;
        }
        return false;
    }

}

File: AndroidCinema/app/src/main/java/com/crpr/androidcinema/presentation/common/ConnectionUtils.kt
package com.crpr.androidcinema.presentation.common

import android.content.Context
import android.net.ConnectivityManager

/**
 * Created by DEATH_STAR on 21/01/2017.
 */

object ConnectionUtils {

    fun isConnectionAvailable(context: Context): Boolean {
        val networkTypes = intArrayOf(ConnectivityManager.TYPE_MOBILE, ConnectivityManager.TYPE_WIFI)
        try {
            val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            for (networkType in networkTypes) {
                val activeNetworkInfo = connectivityManager.activeNetworkInfo
                if (activeNetworkInfo != null && activeNetworkInfo.type == networkType)
                    return true
            }
        } catch (e: Exception) {
            return false
        }

        return false
    }

}
